client/dashboard/sites/index.tsx

```diff
@@ -66,10 +67,27 @@ export default function Sites() {
 	const { data: filteredData, paginationInfo } = filterSortAndPaginate( sites ?? [], view, fields );
 	const [ isModalOpen, setIsModalOpen ] = useState( false );
 
+	const hasFilterOrSearch = ( view.filters && view.filters.length > 0 ) || view.search;
+
+	const emptyHeading = hasFilterOrSearch ? __( 'No sites found' ) : __( 'No sites' );
```

Comment:
When choosing the copy to show, having some sort of filter takes precedence over having no sites. That might be counter intuitive.

My first instinct was that "No sites" should take precedence. If a user has zero sites and they type in a search, surely the more important message is that the user has no sites.

But is a little more nuanced. By default we do not show "hidden" sites. But then when the user starts typing, we assume that maybe they're looking for a specific hidden site and so start showing them.

So when the user sees the "No sites" message, we don't actually know whether the user has literally no sites 😳

And when the user starts searching, they might be looking for a hidden site, and so it is better to show the message "No sites found".